jcl/src/openj9.dtfj/share/classes/com/ibm/dtfj/javacore/parser/j9/section/memory/MemorySectionParser.java

```diff
@@ -59,16 +66,54 @@ protected void topLevelRule() throws ParserException {
 	 */
 	private void memInfo() throws ParserException {
 
-		IAttributeValueMap results = null;
+		IAttributeValueMap results;
+
+		while ((results = processTagLineOptional(T_1STHEAPTYPE)) != null) {
+			// The heap type
+			String segName = results.getTokenValue(MEMORY_SEGMENT_NAME);
+			IAttributeValueMap resultsSpace;
+			while ((resultsSpace = processTagLineOptional(T_1STHEAPSPACE)) != null) {
+				long spaceid = resultsSpace.getLongValue(MEMORY_SEGMENT_ID);
+				String spacetype = resultsSpace.getTokenValue(MEMORY_SEGMENT_TYPE).trim();
+				String spaceName;
+				if (spaceid != IBuilderData.NOT_AVAILABLE) {
+					spaceName = segName + " " + spacetype + " 0x" + Long.toHexString(spaceid);
+				} else {
+					spaceName = segName + " " + spacetype;
+				}
+				JavaHeap heap = fJavaRuntimeBuilder != null ? fJavaRuntimeBuilder.addJavaHeap(spaceName) : null;
+				// Each segment
+				IAttributeValueMap resultsRegion;
+				while ((resultsRegion = processTagLineOptional(T_1STHEAPREGION)) != null) {
+					long id = resultsRegion.getLongValue(MEMORY_SEGMENT_ID);
+					long head = resultsRegion.getLongValue(MEMORY_SEGMENT_HEAD);
+					long size = resultsRegion.getLongValue(MEMORY_SEGMENT_SIZE);
+					long tail = resultsRegion.getLongValue(MEMORY_SEGMENT_TAIL);
+					String type = resultsRegion.getTokenValue(MEMORY_SEGMENT_TYPE).trim();
+					String name;
+					if (id != IBuilderData.NOT_AVAILABLE) {
+						name = type + " 0x" + Long.toHexString(id);
+					} else {
+						name = type;
+					}
+					if (fImageAddressSpaceBuilder != null) {
+						ImageSection section = fImageAddressSpaceBuilder.addImageSection(name, head, size);
+						if (fJavaRuntimeBuilder != null && heap != null) {
```

Comment:
`heap != null` implies `fJavaRuntimeBuilder != null` so the latter is redundant

Reply:
It's existing code, but there are missing checks on lines 135, 145 and 156 for `fImageAddressSpaceBuilder != null`.

Reply:
Fixed